lms/views/basic_lti_launch.py

```diff
@@ -47,6 +47,27 @@ def __init__(self, context, request):
         # Configure the front-end mini-app to run.
         self.context.js_config.update({"mode": "basic-lti-launch"})
 
+        # Add config used by frontend to call `record_submission` API.
+        params = self.request.params
+        if (
+            # The outcome reporting params are typically only available when
+            # students (not teachers) launch an assignment.
+            params.get("lis_result_sourcedid")
+            and params.get("lis_outcome_service_url")
+            # This feature is initially restricted to Canvas.
+            and params.get("tool_consumer_info_product_family_code") == "canvas"
+            and request.feature("speedgrader")
+        ):
+            self.context.js_config["submissionParams"] = {}
+
+            self._set_submission_param("h_username", context.h_username)
+            self._set_submission_param(
+                "lis_result_sourcedid", params.get("lis_result_sourcedid")
+            )
+            self._set_submission_param(
+                "lis_outcome_service_url", params.get("lis_outcome_service_url")
+            )
```

Comment:
These 20 lines of code feel to me like a candidate for extraction into a separate method with its own unit tests

Reply:
I'd entertain suggestions of what that might look like, but I'm honestly not sure that I see the value here.